### PR TITLE
Refacored

### PR DESCRIPTION
**Pull request by AlexGrytsai**

### CODE REVIEW

File: app/fight_knights.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class KnightFighting:
+
+    @staticmethod
+    def battle(knights: dict[object, Knight], fights: dict) -> None:
+        for fighter_1, fighter_2 in fights.items():
+            fighter_1 = knights[fighter_1]
+            fighter_2 = knights[fighter_2]
+            fighter_1.hp -= (fighter_2.power - fighter_1.protection)
+            fighter_2.hp -= (fighter_1.power - fighter_2.protection)
+            if fighter_1.hp < 0:
```
Comment (vasylhnatiuk) [inline app/fight_knights.py:13]: try to avoid duplicating

File: app/knights.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    def knight_armour(self, armours: list[dict]) -> None:
+        for armour in armours:
+            self.armour.append(armour["part"])
+            self.protection += armour["protection"]
+
+    def knight_potion(self, potion: dict) -> None:
+        self.potion = potion["name"]
+        for effect, value in potion["effect"].items():
+            if effect == "power":
```
Comment (vasylhnatiuk) [inline app/knights.py:21]: use settatr instead of duplicating

File: app/main.py [hunk 3]
```diff
@@ -85,130 +88,21 @@
...
-        lancelot["name"]: lancelot["hp"],
-        arthur["name"]: arthur["hp"],
-        mordred["name"]: mordred["hp"],
-        red_knight["name"]: red_knight["hp"],
-    }
+def battle(knights: dict) -> dict:
+    for knight in knights.values():
+        Knight(name=knight["name"], power=knight["power"], hp=knight["hp"],
+               weapon=knight["weapon"])
+        if len(knight["armour"]) != 0:
```
Comment (vasylhnatiuk) [inline app/main.py:98]: bettter move it to Knight `__init__`

File: app/main.py [hunk 4]
```diff
@@ -85,130 +88,21 @@
...
+            Knight.knight_armour(Knight.knights[knight["name"]],
+                                 knight["armour"])
+        if knight.get("potion"):
```
Comment (vasylhnatiuk) [inline app/main.py:101]: same with it

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/knights.py (hunk 2 above)
Comment (EdAlekseiev) [reply app/knights.py:25]: Strill not fixed

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**